File: week4/app/src/main/java/com/example/fragmentjava/AddFragment.java
package com.example.fragmentjava;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.example.fragmentjava.databinding.AddFragmentBinding;

public class AddFragment extends Fragment {

    // Declaration of the addBinding variable with AddFragmentBinding data type
    private AddFragmentBinding addBinding;

    // Constructor for the AddFragment class
    public AddFragment(){}

    // onCreateView() method, called when the view is created
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the View for this fragment
        addBinding = AddFragmentBinding.inflate(inflater, container, false);

        // Create a View object from the inflated binding
        View view = addBinding.getRoot();

        // Set up an onClickListener for the addButton
        addBinding.addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Get the message string from editText
                String message = addBinding.editText.getText().toString();

                // If the message is not empty
                if (!message.isEmpty() ) {
                    // Store the message in SharedPreferences with a key "message"
                    SharedPreferences sharedPref= requireActivity().
                            getSharedPreferences("Message", Context.MODE_PRIVATE);
                    SharedPreferences.Editor spEditor = sharedPref.edit();
                    spEditor.putString("message", message);
                    spEditor.apply();
                }
            }
        });

        // Set up an onClickListener for the clearButton
        addBinding.clearButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Clear the editText view
                addBinding.editText.setText("");
            }
        });

        // Return the View object
        return view;
    }

    // onDestroyView() method, called when the fragment's view is destroyed
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        // Set addBinding variable to null to avoid memory leaks
        addBinding = null;
    }
}

File: week4/FragmentKotlin/app/src/main/java/com/example/fragmentkotlin/AddFragment.kt
package com.example.fragmentkotlin

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.fragmentkotlin.databinding.AddFragmentBinding

//  accepting input from the user and then saving that input to SharedPreferences
class AddFragment : Fragment() {
    // Declare a private nullable variable to hold the ViewBinding.
    private var _binding: AddFragmentBinding? = null

    // Create a binding property that returns the binding or throws an exception if it's null.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the AddFragmentBinding and assign it to _binding.
        _binding = AddFragmentBinding.inflate(inflater, container, false)

        // Get the SharedPreferences instance for the "Message" key.
        val sharedPref: SharedPreferences = requireActivity(). getSharedPreferences(
            "Message",
            Context.MODE_PRIVATE
        )

        // Set an OnClickListener on the addButton to store the message in SharedPreferences.
        binding.addButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View?) {
                // Get the message from the editText.
                val message: String = binding.editText.text.toString()

                // If the message is not empty, save it to SharedPreferences.
                if (message.isNotEmpty()) {
                    val spEditor = sharedPref.edit()
                    spEditor.putString("message", message)
                    spEditor.apply()
                }
            }
        })

        // Set an OnClickListener on the clearButton to clear the editText.
        binding.clearButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View?) {
                binding.editText.setText("")
            }
        })

        // Return the root view of the binding.
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()

        // Set the _binding variable to null to release the ViewBinding.
        _binding = null
    }
}
